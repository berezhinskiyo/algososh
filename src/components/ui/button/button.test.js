import { Button } from "./button";
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

describe('Тестирование кнопки', () => {
    it('кнопки с текстом', () => {
        const tree = renderer
            .create(<Button text="Ок" />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('кнопки без текста', () => {
        const tree = renderer
            .create(<Button />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('заблокированной кнопки', () => {
        const tree = renderer
            .create(<Button disabled />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });


    it('кнопки с индикацией загрузки', () => {
        const tree = renderer
            .create(<Button isLoader />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Проверяем корректность вызова колбека при клике на кнопку', () => {
        window.alert = jest.fn();

        render(<Button text="Ok" onClick={() => {
            alert('Работает')
        }} />)

        const link = screen.getByText("Ok");

        fireEvent.click(link);
        expect(window.alert).toHaveBeenCalledWith('Работает');
    });
});