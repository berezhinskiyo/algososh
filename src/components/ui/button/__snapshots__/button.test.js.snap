// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[` кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[` кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[` кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Ок
  </p>
</button>
`;

exports[`Тестирование  заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование  кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование  кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование  кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Ок
  </p>
</button>
`;

exports[`Тестирование кнопки заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование кнопки кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование кнопки кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Ок
  </p>
</button>
`;

exports[`заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`кнопки с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Ок
  </p>
</button>
`;
